### PR TITLE
Climate IR LG - Support fan only mode and all "on" commands

### LINKED ISSUE #1302: Climate IR LG - Support fan only mode and all "on" commands
**Describe the problem you have/What new integration you would like**

I have an LG central air conditioner unit with a PREMTB00 wall remote control. I found that there are a couple of things not currently supported by the existing implementation:
1. Fan only mode
2. "On" commands to start the air conditioner in a mode different than cool

**Please describe your use case for this integration and alternatives you've tried:**

I created for my specific use case an external component cloning and modifying the current one. I'd like to share my changes and create a pull request, but I don't want to brake existing use cases.

Do all LG units support fan only mode? If not, I was thinking about making it available in the config as a parameter, but considering the value is passed in the constructor of the parent class I reached a dead end. Not proficient on C++ at all.

Any help is greatly appreciated.

**Additional context**

No additional comment

### PR DESCRIPTION
**Pull request by danieldabate**
# What does this implement/fix?

It adds to the Climate IR LG component:
- Fan only mode
- "On" commands to turn the air conditioner in any mode (instead of turning in cold and having to change next)
- Small refactor to make code more straightforward

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes https://github.com/esphome/feature-requests/issues/1302

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** NA

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [x] ESP8266

## Example entry for `config.yaml`:
```yaml
remote_transmitter:
  pin: GPIO14
  carrier_duty_percent: 50%

climate:
  - platform: climate_ir_lg
    name: Air Conditioner
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by jacekk015** [comment]
@OttoWinter sorry for direct mentioning but it's a something that could me merged ad-hoc.
Tested on my two LG air conditioners.

**Comment by jacekk015** [comment]
...

**Comment by danieldabate** [comment]
...

**Comment by danieldabate** [comment]
…

**Comment by jameswood** [comment]
This PR would be helpful! 👍🏻

**Comment by jacekk015** [comment]
Hello!!

**Comment by jameswood** [comment]
> There hasn't been any activity on this pull request recently. This pull request has been automatically marked as stale because of that and will be closed if no further activity occurs within 7 days. Thank you for your contributions.

Does anyone know how we can be helpful here? By "we," I mean those of us who would love to see this merged. Is there anything practical we can do to help?

**Comment by goopaga** [comment]
I'm also lacking the "FAN ONLY" mode in my LG setup and would be great to have this PR merged.
Is there anyone willing to review the code ? Would be lovely.

@danieldabate any clue on why hasn't been integrated yet ? I'm not a coder and a very basic GitHub user :-)

**Comment by danieldabate** [comment]
> I'm also lacking the "FAN ONLY" mode in my LG setup and would be great to have this PR merged.
> 
> Is there anyone willing to review the code ? Would be lovely.
> 
> 
> 
> @danieldabate any clue on why hasn't been integrated yet ? I'm not a coder and a very basic GitHub user :-)

Open source project are complex to manage, with most people participating in their free time. Like me. 

So it's no surprise that maintainers might focus on more important or urgent features. This is just one small feature in a specific component of many available.

I lost hope that this will get merged, because it got buried after 2 years. You could try creating an issue asking for the feature and pointing to this PR to get attention.

Or like the rest of us, download the modified component code and add it as a custom component 😅

**Comment by goopaga** [comment]
> > I'm also lacking the "FAN ONLY" mode in my LG setup and would be great to have this PR merged.
> > Is there anyone willing to review the code ? Would be lovely.
> > @danieldabate any clue on why hasn't been integrated yet ? I'm not a coder and a very basic GitHub user :-)
> 
> Open source project are complex to manage, with most people participating in their free time. Like me.
> 
> So it's no surprise that maintainers might focus on more important or urgent features. This is just one small feature in a specific component of many available.
> 
> I lost hope that this will get merged, because it got buried after 2 years. You could try creating an issue asking for the feature and pointing to this PR to get attention.
> 
> Or like the rest of us, download the modified component code and add it as a custom component 😅

I understand. Thanks for your work and advice. I'll try to integrate your custom component !

**Comment by goopaga** [comment]
Imported the external component and it works flawlessly with an LG ARTCool (year 2004) !
Kudos to @danieldabate !! Well done.

**Comment by danieldabate** [comment]
Awesome! Love to hear it worked also for your model 😄

File: esphome/components/climate_ir_lg/climate_ir_lg.h [hunk 1]
```diff
@@ -12,7 +12,7 @@ const uint8_t TEMP_MAX = 30;  // Celsius
 class LgIrClimate : public climate_ir::ClimateIR {
  public:
   LgIrClimate()
-      : climate_ir::ClimateIR(TEMP_MIN, TEMP_MAX, 1.0f, true, false,
+      : climate_ir::ClimateIR(TEMP_MIN, TEMP_MAX, 1.0f, true, true,
```
Comment (ssieb) [inline esphome/components/climate_ir_lg/climate_ir_lg.h:15]: What does this change do?

Comment (danieldabate) [reply esphome/components/climate_ir_lg/climate_ir_lg.h:15]: Enables the fan only mode. You can check the constructor of ClimateIR class for reference.

**Review by ssieb — APPROVED** [review]

**Comment by danieldabate** [comment]
Thanks @ssieb for the help getting it merged! 🙌

**Pull request merged**